server/src/test/java/org/opensearch/action/support/clustermanager/TransportClusterManagerNodeActionTests.java

```diff
@@ -568,4 +572,124 @@ public void testDelegateToClusterManagerOnNodeWithDeprecatedMasterRole() throws
+                .version(randomIntBetween(0, 10))
+        );
+
+        PlainActionFuture<Response> listener = new PlainActionFuture<>();
+        TransportClusterManagerNodeAction action = new Action("internal:testAction", transportService, clusterService, threadPool);
+        action.execute(request, listener);
+
+        CapturingTransport.CapturedRequest[] capturedRequests = transport.getCapturedRequestsAndClear();
+        assertThat(capturedRequests.length, equalTo(1));
+        CapturingTransport.CapturedRequest capturedRequest = capturedRequests[0];
+        assertTrue(capturedRequest.node.isMasterNode());
+        assertThat(capturedRequest.request, equalTo(request));
+        assertThat(capturedRequest.action, equalTo("internal:testAction"));
+        transport.handleRemoteError(
+            capturedRequest.requestId,
+            new MasterTaskThrottlingException("Throttling Exception : Limit exceeded for test")
+        );
+
+        assertFalse(listener.isDone());
+
+        // waiting for retry to trigger
+        Thread.sleep(100);
+
+        // Retry for above throttling exception
+        capturedRequests = transport.getCapturedRequestsAndClear();
+        assertThat(capturedRequests.length, equalTo(1));
+        capturedRequest = capturedRequests[0];
+        Response response = new Response();
+        transport.handleResponse(capturedRequest.requestId, response);
+
+        assertTrue(listener.isDone());
+        listener.get();
+    }
+
+    public void testRetryForDifferentException() throws InterruptedException, BrokenBarrierException {
+        Request request = new Request();
+        PlainActionFuture<Response> listener = new PlainActionFuture<>();
+        AtomicBoolean exception = new AtomicBoolean(true);
+        AtomicBoolean retried = new AtomicBoolean(false);
+        CyclicBarrier barrier = new CyclicBarrier(2);
+        setState(clusterService, ClusterStateCreationUtils.state(localNode, localNode, new DiscoveryNode[] { localNode }));
+
+        TransportClusterManagerNodeAction action = new Action("internal:testAction", transportService, clusterService, threadPool) {
+            @Override
+            protected void masterOperation(Task task, Request request, ClusterState state, ActionListener<Response> listener)
+                throws Exception {
+                if (exception.getAndSet(false)) {
+                    throw new Exception("Different excetion");
+                } else {
+                    // If called second time due to retry, throw exception
+                    retried.set(true);
+                    throw new AssertionError("Should not retry for other exception");
+                }
+            }
+        };
+        action.execute(request, listener);
+
+        assertFalse(retried.get());
+        assertFalse(exception.get());
+    }
+
+    public void testShouldRetry() {
```

Comment:
there is no action.execute? how is it checking shouldRetry?

Reply:
I started writing this one and then realized shouldRetry is being tested in other UT, so it was redundant UT. Missed to remove it. Will Remove it. 


ShouldRetry is being tested in `testThrottlingRetryLocalMaster` and `testThrottlingRetryRemoteMaster`. Where we are throwing Throttling exception directly and via transport exception as well and verifying that retries are being made.

Reply:
ok thanks.